docs/developers/guides/gas/gas-fees.mdx

```diff
@@ -36,12 +36,73 @@ and gas used.
+object called `extraData`. The two most relevant values here are:
+- A `FIXED_COST` of 0.03 Gwei, which reflects infrastructure costs, and;
+- `VARIABLE_COST`, which is the cost per byte of data submitted to L1. 
+
+These values are inserted into each block's `extraData` field—32 bytes available in every Ethereum 
+block to store any data—and then used by the sequencer and Linea Besu nodes running the correct 
+plugins to expose `linea_estimateGas`, in both cases with the same logic.
+
+Variable cost is calculated with the following formula:
+
+```python
+VARIABLE_COST (4 bytes) = min(max(
+(
+  (
+    ((averageWeightedBaseFee + averageWeightedPriorityFee) * 
+    blob-submission-expected-execution-gas + averageWeightedBlobBaseFee * expected-blob-gas
+  ) / bytes-per-data-submission) * profit-margin
+)
+, min-bound), max-bound)
+```
+
+The `profit-margin` is `3`, ensuring sustainable network profitability. `min-bound` and `max-bound` 
+are variable, and guarantee the gas price stays within a reasonable range.
+
+The variable cost formula enables `linea_estimateGas` to price according to the variable costs of 
+submitting blob data to L1, working out the per-byte cost of that data. The amount of the blob data 
+in each block stays roughly consistent, though the amount _per transaction_ varies, so the 
+`linea_estimateGas` API accounts for this, and ensures a gas price is returned for the transaction 
+that reflects the amount of data it contains. In turn, it ensures that transactions are profitable. 
+
+To determine the priority fee per gas, `linea_estimateGas` takes the previous block's `VARIABLE_COST` 
+into account:
+```python
+min-gas-price = previousBlock.extraData.variable_cost
+baseFeePerGas = vanillaProtocolBaseFee
+priorityFeePerGas = MINIMUM_MARGIN * (min-gas-price * L2_compressed_tx_size_in_bytes / L2_tx_gas_used + extraData.fixed_cost)
+```
 
-You can use `eth_gasPrice` or `eth_feeHistory` to get the gas price, in wei, and you can use 
-`eth_estimateGas` to find out how many units of gas a specific transaction will need. 
+Where:
+- `extraData.variable_cost` is where the previous block's `VARIABLE_COST` is stored
+block
+- `MINIMUM_MARGIN` varies depending on the stage of the transaction:
+  - RPC method, i.e. calling `linea_estimateGas`: `1.2`
+  - In the transaction pool: `0.8`
+  - At transaction selection stage: `1.0`
+
+:::note
+The RPC method and transaction pool values are configurable by node runners according to preference;
+the transaction selection stage value is fixed. For example, it may be preferable to set a lower
+margin to facilitate lower gas prices, but this risks transactions not being included.
+:::
+
+The `linea_estimateGas` formula above essentially ensures that transactions are checked for 
+profitability for inclusion in a block. However, the nonce order of transactions submitted from the 
+same account takes precedence, so there may be occasions where less profitable transactions must be
+included first. 
```

Comment:
```suggestion
The `linea_estimateGas` formula above ensures that transactions are checked for 
profitability for inclusion in a block. However, the nonce order of transactions submitted from the 
same account takes precedence, so there may be occasions where less profitable transactions must be
included first. 
```

Reply:
To clarify on this part a bit more. All transactions will be checked by Sequencer for profitability, no matter the source and how they were priced, `linea_estimateGas` doesn't ensure that transactions are checked for profitability by Sequencer. `linea_estimateGas` will run the same profitability logic as Sequencer and will suggest fees that will make the input transaction profitable